shared/contexts/DFXAPIContextProvider.tsx

```diff
@@ -151,70 +183,79 @@ export function DFXAPIContextProvider (props: PropsWithChildren<{}>): JSX.Elemen
 
   // check if Web session is expired
   const isSessionExpired = async (): Promise<boolean> => {
-    const session = await AuthService.Session
+    const session = await AuthService.Session()
     return session.isExpired
   }
 
+  async function signMessageNoPersistence (message: string): Promise<string> {
+    return await createSignature(debouncedAddress, message)
+  }
+
   // start signing process and return signature
-  const createSignature = async (address: string): Promise<void> => {
+  const createSignature = async (address: string, message?: string): Promise<string> => {
     /**
      * Signing message callback
      * @param provider
      * @returns Promise<Buffer>
      */
-    const signMessage = async (provider: WalletHdNodeProvider<MnemonicHdNode>): Promise<Buffer> => {
+    const signMessage = async (provider: WalletHdNodeProvider<MnemonicHdNode>, altMessage?: string): Promise<Buffer> => {
       const activeIndex = await WalletAddressIndexPersistence.getActive()
       const account = initJellyfishWallet(provider, network, whaleApiClient).get(activeIndex)
       const privKey = await account.privateKey()
       const messagePrefix = getJellyfishNetwork(network).messagePrefix
-      const message = `By signing this message, you confirm that you are the sole owner of the provided DeFiChain address and are in possession of its private key. Your ID: ${address}`
+      const message = altMessage ?? `By signing this message, you confirm that you are the sole owner of the provided DeFiChain address and are in possession of its private key. Your ID: ${address}`
       .split(' ')
       .join('_')
       const signMessage = await getSignMessage(address)
       // signMessage: login with AltCoins possible || message: -> fallback
-      return await signAsync(signMessage ?? message, privKey, true, messagePrefix)
+      return await signAsync((altMessage != null) ? message : signMessage ?? message, privKey, true, messagePrefix)
     }
 
     // message signed callback
-    const onMessageSigned = async (sigBuffer: Buffer): Promise<void> => {
+    const onMessageSigned = async (sigBuffer: Buffer, message?: string): Promise<string> => {
       const sig = sigBuffer.toString('base64')
+      if (message != null) {
+        // console.log('sig: ', sig)
+        return sig
+      }
       await DFXPersistence.setPair({
         network: networkName,
         addr: address,
         signature: sig
       })
       await DFXPersistence.resetPin()
+      return sig
```

Comment:
Does the pin reset still work correctly?

Reply:
why shouldn't it?